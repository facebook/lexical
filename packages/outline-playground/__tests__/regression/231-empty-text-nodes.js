/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import {
  initializeE2E,
  assertHTMLSnapshot,
  assertSelection,
  repeat,
} from '../utils';

describe('Regression test #231', () => {
  initializeE2E({chromium: true, webkit: true, firefox: true}, (e2e) => {
    it(`Does not generate segment error when editing empty text nodes`, async () => {
      const {page} = e2e;
      await page.focus('div.editor');
      await page.keyboard.type('#foo');
      await page.waitForSelector('.editor-text-hashtag');
      await repeat(4, async () => {
        await page.keyboard.press('ArrowLeft');
      });
      await page.keyboard.type('a');
      await page.keyboard.press('Backspace');
      await repeat(4, async () => {
        await page.keyboard.press('ArrowRight');
      });
      await repeat(5, async () => {
        await page.keyboard.press('Backspace');
      });
      await assertHTMLSnapshot(page);
      await assertSelection(page, {
        anchorPath: [0, 0, 0],
        anchorOffset: 0,
        focusPath: [0, 0, 0],
        focusOffset: 0,
      });
    });
  });
});
