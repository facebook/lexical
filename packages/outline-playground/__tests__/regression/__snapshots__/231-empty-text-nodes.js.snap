// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Regression test #231 Chromium Does not generate segment error when editing empty text nodes 1`] = `"<p class=\\"editor-paragraph\\"><span>﻿</span></p><div contenteditable=\\"false\\" class=\\"editor-placeholder\\">Enter some rich text...</div>"`;

exports[`Regression test #231 Firefox Does not generate segment error when editing empty text nodes 1`] = `"<p class=\\"editor-paragraph\\"><span>﻿</span></p><div class=\\"editor-placeholder\\" contenteditable=\\"false\\">Enter some rich text...</div>"`;

exports[`Regression test #231 Webkit Does not generate segment error when editing empty text nodes 1`] = `"<p class=\\"editor-paragraph\\"><span>﻿</span></p><div contenteditable=\\"false\\" class=\\"editor-placeholder\\">Enter some rich text...</div>"`;
